haaska.py

```diff
@@ -328,6 +391,29 @@ def set_percentage(self, val):
         vol = val / 100.0
         self._call_service('media_player/volume_set', {'volume_level': vol})
 
+class ClimateEntity(Entity):
+    def turn_on(self):
+        self._call_service('climate/set_away_mode', {'away_mode': False})
+
+    def turn_off(self):
+        self._call_service('climate/set_away_mode', {'away_mode': True})
+
+    def get_state(self):
```

Comment:
I agree, get state should move up to Entity. In the climate entity I would then maybe include get_min_temp() and get_max_temp() and get_mode(), to make it cleaner in the handlers for the temperature requests. 
There is also some duplication for the range check, for setTemperature and for adjTemperature. Maybe it's better to just add a "is_valid_temp()" to the ClimateEntity.